src/app.module.ts

```diff
@@ -1,9 +1,17 @@
 import { Module } from '@nestjs/common';
 import { AppController } from './app.controller';
 import { AppService } from './app.service';
+import { ConfigModule } from '@nestjs/config';
+import { TypeOrmModule } from '@nestjs/typeorm';
+import { ormConfig } from './config/ormconfig';
 
 @Module({
-  imports: [],
+  imports: [
+    ConfigModule.forRoot({
+        isGlobal: true,
+    }),
+    TypeOrmModule.forRoot(ormConfig),
```

Comment:
W przypadku TypeOrm poszedłbym w konfiguracje asynchroniczną. Dodatkowo super by wyglądało, wydzielić bazę danych do oddzielnego modułu.
Linki warte na spojrzenie:
https://docs.nestjs.com/techniques/database#async-configuration

Tutaj mój przykład rozwiązania tego tematu: 
https://github.com/Przekol/realty-zone-backend/tree/main/src/providers/database
https://github.com/Przekol/realty-zone-backend/blob/main/src/config/typeorm.config.ts
https://github.com/Przekol/realty-zone-backend/blob/main/src/config/env-validation.config.ts
https://github.com/Przekol/realty-zone-backend/blob/main/.env.example

Reply:
Wydaje mi się, że zrobiłem to właśnei według dokumentacji. 
U siebie w pliku envValidationObjectSchema zastsowałeś jakieś sztuczki:
POSTGRES_USER: Joi.string().required(), - nawet nie mam pojęcia o co chodzi...
POSTGRES? - u siebie zastosowałeś bazę Postges? 
Zupełnei nie wiem o co chodz... 

Jeśli masz lepsze rozwiązanie i bedziesz to ogarniał - jasne zajmij się tym.